client/client.go

```diff
@@ -353,6 +355,31 @@ func (c *Client) Node() *structs.Node {
 	return c.config.Node
 }
 
+func (c *Client) FSList(allocID string, path string) ([]*allocdir.AllocFileInfo, error) {
```

Comment:
Then the client can have a method called `GetAllocFS(allocID string) (AllocDirFS, error)`. This lets you remove the duplicate code from the client/alloc_runner